dll/win32/uxtheme/errinfo.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * PROJECT:     ReactOS uxtheme.dll
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Error information of UXTHEME
+ * COPYRIGHT:   Copyright 2025 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+
+#include "uxthemep.h"
+#include <stdlib.h>
+#include <strsafe.h>
+
+HRESULT
+UXTHEME_MakeErrorLast(VOID)
+{
+    return UXTHEME_MakeError32(GetLastError());
+}
+
+PTMERRINFO
+UXTHEME_GetParseErrorInfo(_In_ BOOL bCreate)
+{
+    PTMERRINFO pErrInfo;
+
+    if (gdwErrorInfoTlsIndex == TLS_OUT_OF_INDEXES)
+        return NULL;
+
+    pErrInfo = TlsGetValue(gdwErrorInfoTlsIndex);
+    if (pErrInfo)
+        return pErrInfo;
+
+    if (bCreate)
+    {
+        pErrInfo = LocalAlloc(LPTR, sizeof(TMERRINFO));
+        TlsSetValue(gdwErrorInfoTlsIndex, pErrInfo);
+    }
+
+    return pErrInfo;
+}
+
+VOID
+UXTHEME_DeleteParseErrorInfo(VOID)
+{
+    PTMERRINFO pErrInfo = UXTHEME_GetParseErrorInfo(FALSE);
+    if (!pErrInfo)
+        return;
+
+    TlsSetValue(gdwErrorInfoTlsIndex, NULL);
+    LocalFree(pErrInfo);
+}
+
+static BOOL
+UXTHEME_FormatLocalMsg(
+    _In_ HINSTANCE hInstance,
+    _In_ UINT uID,
+    _Out_ LPWSTR pszDest,
+    _In_ SIZE_T cchDest,
+    _In_ LPCWSTR pszDrive,
+    _In_ PTMERRINFO pErrInfo)
+{
+    WCHAR szFormat[MAX_PATH];
+    LPWSTR pch;
+
+    if (!LoadStringW(hInstance, uID, szFormat, _countof(szFormat)))
+        return FALSE;
+
+    // Convert "%1" and "%2" to "%s"
```

Comment:
What about not changing that (what happens if there is %3 used?), but using `FormatMessageW` instead?
Because the %1, etc are placement indicators, so if you have a localized string like this: `foo %2 bar %1` where it is intended (in the syntax of the language) that the "path0" of your error goes where %1 is (and "path1" where %2 is, and not the other way around), by now replacing %1 and %2 with %s, you're breaking the formatting.

Reply:
Used `FormatMessageW` and don't use conversion.